libs/components/core/src/lib/modules/resize-observer/resize-observer.service.ts

```diff
@@ -24,98 +61,102 @@ type ResizeObserverTracking = {
+  readonly #ngUnsubscribe = new Subject<void>();
+  readonly #resizeObserver = new ResizeObserver((entries) =>
+    this.#resizeSubject.next(entries)
+  );
+  readonly #resizeSubject = new Subject<ResizeObserverEntry[]>();
+  readonly #tracking = new Map<Element, Observable<ResizeObserverEntry>>();
+  readonly #window = inject(SkyAppWindowRef);
+  readonly #zone = inject(NgZone);
+
+  constructor() {
+    this.#expectWindowError();
+    // Because the resize observer is a native browser API, it does not shut down
+    // synchronously when the service is destroyed. Leave the error handling
+    // accommodation in place until the application is destroyed. This also works
+    // for the test runner.
+    inject(ApplicationRef).onDestroy(() => this.#resetWindowError());
+  }
 
   public ngOnDestroy(): void {
-    this.#next.forEach((value) =>
-      this.#window.nativeWindow.cancelAnimationFrame(value)
-    );
-    this.#tracking.forEach((value) => {
-      value.subject.complete();
-      this.#resizeObserver.unobserve(value.element);
-    });
+    this.#ngUnsubscribe.next();
+    this.#ngUnsubscribe.complete();
     this.#resizeObserver.disconnect();
   }
 
   /**
    * Create rxjs observable to get size changes for an element ref.
    */
   public observe(element: ElementRef): Observable<ResizeObserverEntry> {
-    return this.#observeAndTrack(element).subjectObservable;
-  }
-
-  #observeAndTrack(element: ElementRef): ResizeObserverTracking {
-    const checkTracking = this.#tracking.findIndex((value) => {
-      return !value.subject.closed && value.element === element.nativeElement;
-    });
+    const checkTracking = this.#tracking.has(element.nativeElement);
 
-    if (checkTracking === -1) {
-      this.#resizeObserver.observe(element.nativeElement);
+    if (!checkTracking) {
+      this.#tracking.set(
+        element.nativeElement,
+        new Observable<ResizeObserverEntry[]>((observer) => {
+          const subscription = this.#resizeSubject.subscribe(observer);
+          this.#resizeObserver?.observe(element.nativeElement);
+          return () => {
+            this.#resizeObserver?.unobserve(element.nativeElement);
+            subscription.unsubscribe();
+            this.#tracking.delete(element.nativeElement);
+          };
+        }).pipe(
+          filter(Boolean),
+          filter((entries) =>
+            entries.some((entry) => entry.target === element.nativeElement)
+          ),
+          map(
+            (entries) =>
+              entries.find(
+                (entry) => entry.target === element.nativeElement
+              ) as ResizeObserverEntry
+          ),
+          distinctUntilChanged(
+            (a, b) =>
+              Math.round(a.contentRect.width) ===
+                Math.round(b.contentRect.width) &&
+              Math.round(a.contentRect.height) ===
+                Math.round(b.contentRect.height)
+          ),
+          shareReplay({ bufferSize: 1, refCount: true }),
```

Comment:
Replay the last value for late subscribers and track all subscribers so it shuts down when there are no more references.